### PR TITLE
feat(UI-1021): add feat close modal on Esc

### PR DESCRIPTION
**Pull request by J1za**
## Description
When the modal opens, allow closing it by clicking Esc on the keyboard.
## Linear Ticket
https://linear.app/autokitteh/issue/UI-1021/close-modal-on-click-on-esc
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/molecules/modal.tsx [hunk 1]
```diff
@@ -49,16 +74,17 @@ export const Modal = ({ children, className, hideCloseButton, name }: ModalProps
 						className={modalClasses}
 						exit="hidden"
 						initial="hidden"
+						ref={modalRef}
 						variants={modalVariants}
 					>
-						{hideCloseButton ? null : (
+						{!hideCloseButton ? (
```
Comment (RonenMars) [inline src/components/molecules/modal.tsx:80]: please return as it was.
The condition shouldn't be with `!`.

File: src/components/molecules/modal.tsx [hunk 2]
```diff
@@ -27,10 +27,35 @@ export const Modal = ({ children, className, hideCloseButton, name }: ModalProps
...
+		if (isOpen && modalRef.current) {
+			const firstInput = modalRef.current.querySelector("input, textarea, select");
+			if (!firstInput) return;
+
+			(firstInput as HTMLElement).focus();
+		}
+
+		const handleKeyDown = (event: KeyboardEvent) => {
+			if (event.key === "Escape" && isOpen) {
+				onClose(name);
```
Comment (RonenMars) [inline src/components/molecules/modal.tsx:47]: I think we should `stopPropagation` after we closed the modal, we don't want close other modals or other elements, but only the current modal.

File: src/components/molecules/modal.tsx (hunk 1 above)
Comment (J1za) [reply src/components/molecules/modal.tsx:80]: reverted

File: src/components/molecules/modal.tsx (hunk 2 above)
Comment (J1za) [reply src/components/molecules/modal.tsx:47]: added

**Review by RonenMars — APPROVED** [review]

**Pull request merged**